earth_enterprise/src/fusion/autoingest/sysman/StateUpdater.cpp

```diff
@@ -372,9 +374,7 @@ void StateUpdater::UpdateWaitingAssets(
   const SharedString ref = version->GetRef();
   const AssetDefs::State newState = version->state;
   waitingListeners.Update(ref, newState, oldState, waitingFor.inputs);
-  if (IsParent(version)) {
```

Comment:
FYI, this is a little code cleanup. The update function won't add the ref to the waiting list unless its waiting for at least one child, so non-parents will never be added to the waiting list anyway.